test/cli/content_blocking_test.go

```diff
@@ -98,6 +101,28 @@ func TestContentBlocking(t *testing.T) {
 		assert.Equal(t, http.StatusOK, resp.StatusCode)
 	})
 
+	// Does the gateway deny indirectly blocked CID?
+	t.Run("Gateway Denies indirectly blocked CID", func(t *testing.T) {
+		t.Parallel()
+		resp := client.Get("/ipfs/" + blockedCIDInParentDirCID)
+		assert.Equal(t, http.StatusGone, resp.StatusCode, statusExpl)
+		assert.NotEqual(t, "directly blocked file content", resp.Body)
+		assert.Contains(t, resp.Body, blockedMsg, bodyExpl)
+	})
+
+	// Does the gateway return CAR
+	t.Run("Gateway returns parent path CAR without blocked CID", func(t *testing.T) {
+		resp := client.Get("/ipfs/" + parentDirCID + "?format=car")
+		require.Equal(t, http.StatusOK, resp.StatusCode)
+
+		bs, err := cbs.NewReadOnly(strings.NewReader(resp.Body), nil)
+		require.NoError(t, err)
+
+		has, err := bs.Has(context.Background(), cid.MustParse(blockedCIDInParentDirCID))
+		require.NoError(t, err)
+		require.False(t, has)
+	})
```

Comment:
Fail on both 0.26 and `master`.

Reply:
Filled https://github.com/ipfs/kubo/issues/10361